File: Watcher/Models/v1/TestExecutions/TestExecution.cs
using System;
namespace Watcher.Models.v1.TestExecutions
{
    public class TestExecution: IdAware
    {
        public DateTime DateTime { get; set; }

        public string Log { get; set; }

        public bool IsSuccessful { get; set; }

        public int TestId { get; set; }
    }
}

File: Watcher/client_app/packages/watcher_client_dal/lib/models/test_execution/test_execution.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import '../ito_json_aware.dart';

part 'test_execution.freezed.dart';
part 'test_execution.g.dart';

@Freezed(
    copyWith: false,
    fromJson: true,
    toJson: true,
    map: FreezedMapOptions(map: false, mapOrNull: false, maybeMap: false),
    when: FreezedWhenOptions(when: false, whenOrNull: false, maybeWhen: false),
    makeCollectionsUnmodifiable: false
)
class TestExecution with _$TestExecution implements IToJsonAware {
  factory TestExecution({
    required int id,
    required DateTime dateTime,
    required String log,
    required bool isSuccessful,
    required int testId
  }) = _$DataTest;

  factory TestExecution.fromJson(Map<String, dynamic> json) => _$TestExecutionFromJson(json);
}